src/Diagnostics.RuntimeHost/Controllers/DevOpsController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Diagnostics.RuntimeHost.Services.DevOpsClient;
+using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Diagnostics.RuntimeHost.Controllers
+{
+    [Authorize]
+    [Produces("application/json")]
+    [Route(UriElements.DevOps)]
+    public class DevOpsController : Controller
+    {
+        IRepoClient _devOpsClient;
+
+        public DevOpsController(IRepoClient devOpsClient)
+        {
+            _devOpsClient = devOpsClient;
+        }
+
+        [HttpPost(UriElements.DevOpsMakePR)]
+        public async Task<IActionResult> MakePullRequestAsync(string resourceURI, [FromBody]JToken jsonBody)
+        {
+            string sourceBranch = jsonBody[$"sourceBranch"].ToString();
+            string targetBranch = jsonBody[$"targetBranch"].ToString();
+            string title = jsonBody[$"title"].ToString();
```

Comment:
This parameter validation in json body is a repeated pattern I see in multiple controllers.
Can you create a utility method to do this.
Something like :

bool ValidateRequestBody(JToken jsonBody, string[] fieldNames, out string outputMessage = null);